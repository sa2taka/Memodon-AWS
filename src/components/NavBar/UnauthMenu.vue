<template>
  <nav-bar-menu>
    <v-list-item>
      <v-list-item-content>
        <v-list-item-title>
          <v-btn text @click="singinWithTwitter">
            <v-icon color="#1DA1F2" v-html="'$vuetify.icons.twitter'"></v-icon>
            <p class="ml-1 my-auto">Sign in with Twitter</p>
          </v-btn>
        </v-list-item-title>
      </v-list-item-content>
    </v-list-item>
    <v-divider></v-divider>
    <v-list-item>
      <template v-slot:default="{ active, toggle }">
        <v-list-item-content>
          <v-list-item-title>Dark Mode</v-list-item-title>
        </v-list-item-content>

        <v-list-item-action>
          <v-switch
            v-model="isDark"
            class="center-switch"
            :dark="isDark"
            id="dark-mode-switch"
          ></v-switch>
        </v-list-item-action>
      </template>
    </v-list-item>
  </nav-bar-menu>
</template>

<script lang="ts">
import { Component, Watch, Vue } from 'vue-property-decorator';
import NavBarMenu from '@/components/NavBar/NavBarMenu.vue';
import theme from '@/store/modules/theme';

@Component({
  components: {
    NavBarMenu,
  },
})
export default class UnauthMenu extends Vue {
  private isDark: boolean = false;

  public created() {
    this.isDark = theme.theme === 'dark';
    this.$store.subscribe((mutation, state) => {
      if (mutation.type === 'theme/setTheme') {
        this.isDark = mutation.payload === 'dark';
      }
    });
  }

  @Watch('isDark')
  public onChangeTheme(isDark: boolean) {
    theme.setTheme(isDark ? 'dark' : 'light');
  }

  private singinWithTwitter() {
    const url = 'https://api.memodon.com/v1/twitter/auth-page';
    location.href = url;
  }
}
</script>

<style lang="scss" scoped></style>
