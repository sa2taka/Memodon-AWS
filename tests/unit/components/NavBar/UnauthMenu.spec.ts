import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
import UnauthMenu from '@/components/NavBar/UnauthMenu.vue';

const localVue = createLocalVue();

localVue.use(Vuetify);

describe('Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(UnauthMenu, { localVue });
  });

  test('is a Vue instance', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
