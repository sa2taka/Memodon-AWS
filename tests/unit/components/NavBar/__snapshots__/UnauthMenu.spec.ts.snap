// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is a Vue instance 1`] = `
<nav-bar-menu-stub>
  <v-list-item-stub
    activeclass=""
    tag="div"
  >
    <v-list-item-content-stub>
      <v-list-item-title-stub>
        <v-btn-stub
          activeclass=""
          tag="button"
          text="true"
          type="button"
        >
          <v-icon-stub
            color="#1DA1F2"
          />
           
          <p
            class="ml-1 my-auto"
          >
            Sign in with Twitter
          </p>
        </v-btn-stub>
      </v-list-item-title-stub>
    </v-list-item-content-stub>
  </v-list-item-stub>
   
  <v-divider-stub />
   
  <v-list-item-stub
    activeclass=""
    tag="div"
  />
</nav-bar-menu-stub>
`;
